packages/fast-components-styles-msft/src/radio/index.ts

```diff
@@ -0,0 +1,88 @@
+import { ComponentStyles, ComponentStyleSheet, ICSSRules } from "@microsoft/fast-jss-manager";
+import { IRadioClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    disabledContrast,
+    ensureNormalContrast,
+    hoverContrast,
+    normalContrast
+} from "../utilities/colors";
+
+/* tslint:disable:max-line-length */
+const styles: ComponentStyles<IRadioClassNameContract, IDesignSystem> = (config: IDesignSystem): ComponentStyleSheet<IRadioClassNameContract, IDesignSystem>  => {
+    const designSystem: IDesignSystem = withDesignSystemDefaults(config);
+    const backgroundColor: string = designSystem.backgroundColor;
+    const foregroundColor: string = ensureNormalContrast(config.contrast, designSystem.foregroundColor, designSystem.backgroundColor);
+    const radioColor: string = normalContrast(designSystem.contrast, foregroundColor, backgroundColor);
+    const radioHover: string = hoverContrast(designSystem.contrast, foregroundColor, backgroundColor);
+    const radioDisabled: string = disabledContrast(
+        designSystem.contrast,
+        foregroundColor,
+        backgroundColor
+    );
+
+    return {
+        radio: {
+            display: "inline-flex",
+            flexDirection: "row",
+            alignItems: "center"
+        },
+        radio_input: {
+            position: "absolute",
+            width: "20px",
+            height: "20px",
+            appearance: "none",
+            borderRadius: "50%",
+            boxSizing: "content-box",
+            margin: "0",
+            zIndex: "1",
+            background: backgroundColor,
+            boxShadow: `inset 0 0 0 1px ${radioColor}`,
+            "&:hover": {
+                boxShadow: `inset 0 0 0 1px ${radioHover}`,
+            },
+            "&:focus": {
+                outline: "none",
+                boxShadow: `inset 0 0 0 2px ${radioColor}`,
+            },
+            "&:checked": {
+                "& + span": {
+                    "&::before": {
+                        position: "absolute",
+                        zIndex: "1",
+                        content: "\"\"",
+                        borderRadius: "50%",
+                        background: radioColor
+                    }
+                }
+            },
+        },
+        radio_stateIndicator: {
+            position: "relative",
+            borderRadius: "50%",
+            display: "inline-block",
+            width: "20px",
+            height: "20px",
+            flexShrink: "0",
+            marginRight: "5px",
+            "&::before": {
+                top: "4px",
+                left: "4px",
+                height: "12px",
+                width: "12px"
+           }
+        },
+        radio__disabled: {
+            cursor: "not-allowed",
+            "& $radio_input": {
+                boxShadow: `inset 0 0 0 1px ${radioDisabled}`
+            },
+            "& $radio_label": {
```

Comment:
does `radio_label` only exist for styling of the disabled radio? I thought we were going with a prop value to style the label as "disabled" separately. Can you add some insight into why we reverted back to adding a class to a child? If a user wants to change the disabled color on the label using JssStylesheet, would they pass it to the radio or the label?

Reply:
looks like I will be using radio_label for RTL. 
But as for why we ended up coming back to adding a class, I will let @nicholasrice answer that.

Reply:
That's fine - just looking for clarity on the change.

Reply:
@chrisdholt either is acceptable, but the recommendation is through the radio component's stylesheet, because the radio provides a single point of entry into the component's state.

The crux of the issue IMO is that a text element can never actually be *disabled* - it can only be related to disabled things. If we were to implement a *disabled* prop, it would be for the sole purpose of applying styling. To me, it makes more sense to contextually apply disabled styles to the label when the context is disabled than it does to add a stateful prop to an otherwise stateless component with un-contextual styles mapped to that prop. 

With a disabled prop on the label, it would be possible to create a non-disabled radio with a disabled label, and disabled radio with a non-disabled label. Additionally, the radio stylesheet wouldn't have the ability to style the label - set aside disabled for a moment, what if I want to apply margin-left to the label from the radio stylesheet?

Reply:
@nicholasrice for sure - when @janechu and I initially discussed this the idea was to have a prop not that distinguished the label as disabled, but the related input as disabled. So, I think we were in agreement in principle on that, perhaps it wasn't relayed in an offline conversation that way. Consider the benefit to disabled text-fields (no label included in base component), checkbox (where label may not be nested and unrelated to the actual input); those scenarios could benefit as well from adding a class on label to distinguish that the input is disabled.

On the others, I'll take a bit of a "devil's advocate" stance on a few things. Don't want to block (and don't plan to), but it seems like some of the points I brought up previously (also offline) may not have been articulated fully on this. 

>With a disabled prop on the label, it would be possible to create a non-disabled radio with a disabled label, and disabled radio with a non-disabled label. Additionally, the radio stylesheet wouldn't have the ability to style the label 

Correct. But, that aligns well with our principle that things that are *bad* are possible. We don't show it as a pattern, and we don't recommend it, but we haven't built out our components to prevent things like this previously.

>  set aside disabled for a moment, what if I want to apply margin-left to the label from the radio stylesheet?

I think this is probably the *best* argument for adding the label class here, but I would still argue that our principle has been to handle this through composition, jssStylesheet, and the many other ways that we expose. In the initial discussion I had on this, my primary point was, if we didn't go the prop route, we should handle adding the class in the MSFT version rather than the base version as it more closely follows an established pattern and our principles.

Reply:
To me, a principle of allowing "bad" things does not justify intentionally exposing more opportunity for failure when solutions that *don't* expose that opportunity exist.

Principally, a component can apply classes arbitrarily - we do this all the time. We use classes to apply styling to component parts, in base classes and out. I don't see how the label - a component part of the radio - is any different.

I understand the argument around text-box. From my perspective, the root of that issue is there is no abstraction to create a composite component with a label and checkbox. If we simply add a disabled prop to label, we've added an additional place an author needs to apply app state, which I think we can both agree is not an ideal and ergonomic solution.

If you feel we should not go this direction, please block the PR and we can figure out the best direction forward.

Reply:
Like I said above, I don't plan on blocking, nor am I currently blocking. We add classes all over; we have not modified children in this way up until this point. I wasn't arguing that as a reason not to do it, just trying to work through the implications.